### PR TITLE
Feature: invoke user profile apis to map messaging user

### PR DESCRIPTION
**Pull request by SamSalvatico**
### Description

- Moved `building-blocks-sdk` to `es-module`
- Invoked `findUser` API in `messaging-api` to search for the imported users

## Type

- [ ] **Dependency upgrade**
- [ ] **Bug fix**
- [x] **New feature**
- [x] **Dev change**

### Checklist:

- [x] This change might impact the developer experience of others and should be communicated

### CODE REVIEW

**Comment by SamSalvatico** [comment]
Hi all!

I had to set `"type": "module"` in `building-blocks-sdk` to make us able to import it in our APIs services
I just want to notify it to you, shouldn't disrupt Next services, I tested it locally also building images and it seems to work as expected

File: packages/building-blocks-sdk/package.json [hunk 1]
```diff
@@ -8,5 +8,7 @@
   "devDependencies": {
     "@types/node": "20.10.4",
     "typescript": "5.3.3"
-  }
+  },
+  "type": "module",
```
Comment (edge33) [inline packages/building-blocks-sdk/package.json:12]: good change, but we should also build this to plain js in the future.

**Review by edge33 — APPROVED** [review]
LGTM

File: packages/building-blocks-sdk/package.json (hunk 1 above)
Comment (SamSalvatico) [reply packages/building-blocks-sdk/package.json:12]: totally agree

**Review by Thurfjell — APPROVED** [review]
Lots of abstractions here. 

I'm always going to advocate to write a bit more boring code, without abstractions if possibly. I don't believe we need a JavaScript utility "findBy" when you can write each individual SQL query quite easily eg. It's always easier to come back and modify when it's just pure simpleton code, repeated or not.

**Pull request merged**